c7n/resources/account.py

```diff
@@ -367,6 +367,43 @@ def process(self, resources, event=None):
         return []
 
 
+@filters.register('access-analyzer')
+class AccessAnalyzer(ValueFilter):
+    """Check for access analyzers in an account
+
+    :example:
+
+    .. code-block:: yaml
+
+      policies:
+        - name: account-access-analyzer
+          resource: account
+          filters:
+            - type: access-analyzer
+              key: 'status'
+              value: ACTIVE
+              op: eq
+    """
+
+    schema = type_schema('access-analyzer', rinherit=ValueFilter.schema)
+    schema_alias = False
+    permissions = ('access-analyzer:ListAnalyzers',)
+
+    def process(self, resources, event=None):
+        account = resources[0]
+        if not account.get('c7n:matched_analyzers'):
```

Comment:
what's the intent on the condition, if matched is defined, and we don't execute the conditional here, the analyzers will be undefined later. typical use account.setdefault('c7n:matched_analyzers', []) .. the previous version had two separate annotations one for the analyzers, one for api cached analyzers and one for a matched analyzer. this version does away with the api cached analyzers, which is fine, but the condition logic here doesn't cleanly handle both condition states.

Reply:
Yep makes sense, that was definitely a bug. I think it should be fixed now - I iterate through each analyzer and if it matches then I add it to matched_analyzers.

Reply:
this use of annotation cache by the filter is actually another bug imo, ie it short-circuits the filter evaluation based on the presence of a previous match. 

also when using an annotation key (c7n:matched-analyzers) repeatedly its better to define as a class variable to avoid typos in usage. 

generally i think we also want to support chaining these filters, ie find active && last analyzed more than 30 days old. 

i'll push an update to address.

Reply:
Makes sense - thanks for the help on this one!